x-pack/plugins/ingest_manager/server/plugin.ts

```diff
@@ -148,6 +150,9 @@ export class IngestManagerPlugin implements Plugin<IngestManagerSetupContract> {
     }
     return deepFreeze({
       esIndexPatternService: new ESIndexPatternSavedObjectService(),
+      agentService: {
```

Comment:
I think we should expose `agentService` during start, it's probably not going to work if you use it during setup

Reply:
Thanks for catching that @nchaulet. Yeah, I believe we're encouraged to put as much as possible in start. Especially anything that could Error.

Reply:
I think this is a setup dependency, because we do the route registration in the setup method which I believe is the place for this action. These are just service definitions are they are not consumed in the setup step but in the handlers.

Reply:
Yes `esIndexPatternService` should probably be moved to start too.
If it's too much refacto for this PR I am ok to have a follow up PR for that.

Reply:
Gotcha, yeah I was talking with @nnamdifrankie about that. Right now endpoint is accessing the ingest contract in the `setup` function because that's where the routes are being registered. If we move the contract to the `start` function I think we'd have to use a singleton or something to point to the ingest contract because I don't think we can register the endpoint routes in the `start` function. It seems like other plugins do this. An example is the encrypted saved objects: https://github.com/elastic/kibana/blob/master/x-pack/plugins/actions/server/plugin.ts#L243

Might be easier to do that in a follow up PR.

Reply:
@nchaulet @jfsiii @jonathan-buttner 
FYI https://github.com/elastic/endpoint-app-team/issues/357